### PR TITLE
Add links to all the new detail edit pages

### PR DESCRIPTION
**Pull request by CrystalPea**
Trello ticket: https://trello.com/c/x5mzD7bS/49-supplier-add-links-to-all-the-new-change-pages

screen shot 2018-03-05 at 16 31 09

### CODE REVIEW

File: app/templates/suppliers/details.html [hunk 1]
```diff
@@ -71,23 +71,28 @@
...
-  {% call summary.row() %}
-    {{ summary.field_name('Registered company name') }}
-      {{ summary.text(supplier.registeredName) }}
-  {% endcall %}
-{% endif %}
+{% call summary.row() %}
+  {{ summary.field_name('Registered company name') }}
+  {% if supplier.registeredName %}
+    {{ summary.text(supplier.registeredName) }}
+    {{ summary.edit_link(label="Change", link=url_for(".edit_supplier_registered_name")) }}
```
Comment (katstevens) [inline app/templates/suppliers/details.html]: Can we use the hidden_text field for these `edit_links` to provide context for screen reader users? For example:

```
summary.edit_link(label="Change", link=url_for(".edit_supplier_registered_name"), hidden_text="Registered company name")
```

See here for more context: https://github.com/alphagov/digitalmarketplace-supplier-frontend/pull/804

File: app/templates/suppliers/details.html [hunk 2]
```diff
@@ -101,48 +106,65 @@
...
-      {{ summary.text(supplier.companiesHouseNumber) }}
-    {% endcall %}
-  {% endif %}
-
-  {% if supplier.otherCompanyRegistrationNumber and not supplier.companiesHouseNumber and country_name != 'United Kingdom' %}
-    {% call summary.row() %}
-      {{ summary.field_name('Registration number') }}
-      {{ summary.text(supplier.otherCompanyRegistrationNumber) }}
-    {% endcall %}
+    {{ summary.edit_link(label="Change", link=url_for(".edit_registered_address")) }}
```
Comment (katstevens) [inline app/templates/suppliers/details.html]: Same here for adding `hidden_text="Registration number"`

**Comment by laurenceberry** [comment]
Looks good from my perspective.

File: app/templates/suppliers/details.html (hunk 1 above)
Comment (CrystalPea) [reply app/templates/suppliers/details.html]: Yes, great idea :)

**Comment by samuelhwilliams** [comment]
Took me way too long to spot this, but your `hidden_text` attributes are nested inside the `url_for` call, so you're generating weird URLs rather than adding the hidden text.

Also I don't think we want to have the verb (e.g. 'change', 'add') included as part of the hidden text, as a screenreader will read the visible text followed by the hidden text, which will end up with e.g. 'Change change registered company name'. This ignores our internationalisation issue for now, but is consistent with other uses of the hidden_text attribute (e.g. as in @katstevens's example).

**Comment by CrystalPea** [comment]
@samuelhwilliams good spot with the hidden url, I probably should not have added those just before going home 😅 . And good point with the verbs, I will get rid of them.

Also, do you think we need hidden_text attributes on both "Change" and "Answer Required" links?
I am asking as Answer Required link is just next to the row title...

**Comment by katstevens** [comment]
The `hidden_text` is needed where there's potential ambiguity. So if there are multiple 'Change' links on the page, the screen reader needs to be able to differentiate between them.

**Comment by CrystalPea** [comment]
@katstevens good idea, I will check with @constancecerf 

As to the change from "or" to "and": yes, it's intentional. In earlier iteration of the page we did not have "Answer required" link so it made sense to show any information that is there. Now, all this information will need to be submitted for a supplier to be able to proceed with their framework application, so if any is missing, an "Answer required" link should be shown.

**Comment by TheDoubleK** [comment]
The links have moved out of line:
screen shot 2018-03-06 at 17 51 24

**Comment by CrystalPea** [comment]
@TheDoubleK thanks for spotting, fixed :) (somehow I am least cautious when I am introducing changes as a part of code review, I need to work on that 😬 )
screen shot 2018-03-07 at 15 38 34

**Comment by CrystalPea** [comment]
@TheDoubleK good suggestion, I changed the copy of these hidden_text attributes now 👍. Merging when Travis is done ☺️

**Pull request merged**